atomic_reactor/utils/resource_manager.py

```diff
@@ -7,53 +7,152 @@
 """
 
 import backoff
-import json
-import jsonschema
 import logging
-import paramiko
-import random
 import os
+import paramiko
+import time
 from contextlib import contextmanager
 from datetime import datetime
+from functools import cached_property
 from shlex import quote
-from typing import List
+from typing import List, Optional, Tuple
+from paramiko.channel import ChannelFile  # just for type annotation
 
-RESOURCE_LOCK_SCHEMA = 'schemas/resource-lock-content.json'
-SSH_COMMAND_TIMEOUT = 45
-TIMEOUT_FAIL_BUILD = 3000
-RELATIVE_PATH = 'resource_manager'
-RESOURCE_INFO_JSON = 'info.json'
+SSH_COMMAND_TIMEOUT = 30
+SLOTS_RELATIVE_PATH = "osbs_slots"
 RETRY_ON_SSH_EXCEPTIONS = (paramiko.ssh_exception.NoValidConnectionsError,
                            paramiko.ssh_exception.SSHException)
-BACKOFF_FACTOR = 5
+BACKOFF_FACTOR = 3
 MAX_RETRIES = 3
 
 logger = logging.getLogger(__name__)
 
+__all__ = [
+    "RemoteHost",
+    "RemoteHostsPool",
+]
```

Comment:
Should `LockedResource` also be public API? Since that is the return value of pool.lock_resource()?

Reply:
It should, but I don't see a case of using it at this moment, so I didn't export it, I'll add it to list.